File: src/main/java/com/back/pinco/domain/tag/dto/PinTagDto.java
package com.back.pinco.domain.tag.dto;

import com.back.pinco.domain.tag.entity.PinTag;
import java.time.LocalDateTime;

public record PinTagDto(
        Long id,
        Long pinId,
        TagDto tag
) {
    public PinTagDto(PinTag pinTag) {
        this(
                pinTag.getId(),
                pinTag.getPin().getId(),
                new TagDto(pinTag.getTag())
        );
    }
}

File: src/main/java/com/back/pinco/domain/tag/dto/PinTagDto.kt
package com.back.pinco.domain.tag.dto

import com.back.pinco.domain.tag.entity.PinTag

data class PinTagDto(
    val id: Long,
    val pinId: Long,
    val tag: TagDto
) {
    constructor(pinTag: PinTag) : this(
        pinTag.id!!,
        pinTag.pin.id!!,
        TagDto(pinTag.tag)
    )
}
